tck/features/MiscErrorAcceptance.feature

```diff
@@ -0,0 +1,216 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Feature: MiscErrorAcceptance
+
+  Background:
+    Given any graph
+
+  Scenario: Failing on incorrect unicode literal
+    When executing query:
+      """
+      RETURN '\uH'
+      """
+    Then a SyntaxError should be raised at compile time: InvalidUnicodeLiteral
+
+  Scenario: Failing on merging relationship with null property
+    When executing query:
+      """
+      MERGE (a)-[r:X {p: null}]->(b)
+      """
+    Then a SemanticError should be raised at compile time: MergeReadOwnWrites
+
+  Scenario: Failing on merging node with null property
+    When executing query:
+      """
+      MERGE ({p: null})
+      """
+    Then a SemanticError should be raised at compile time: MergeReadOwnWrites
+
+  Scenario: Failing on aggregation in WHERE
+    When executing query:
+      """
+      MATCH (a)
+      WHERE count(a) > 10
+      RETURN a
+      """
+    Then a SyntaxError should be raised at compile time: InvalidAggregation
+
+  Scenario: Failing on aggregation in ORDER BY after RETURN
+    When executing query:
+      """
+      MATCH (n)
+      RETURN n.prop1
+        ORDER BY max(n.prop2)
+      """
+    Then a SyntaxError should be raised at compile time: InvalidAggregation
+
+  Scenario: Failing on aggregation in ORDER BY after WITH
+    When executing query:
+      """
+      MATCH (n)
+      WITH n.prop1 AS foo
+        ORDER BY max(n.prop2)
+      RETURN foo AS foo
+      """
+    Then a SyntaxError should be raised at compile time: InvalidAggregation
+
+  Scenario: Failing when not aliasing expressions in WITH
+    When executing query:
+      """
+      MATCH (a)
+      WITH a, count(*)
+      RETURN a
+      """
+    Then a SyntaxError should be raised at compile time: NoExpressionAlias
+
+  Scenario: Failing when using undefined variable in pattern
+    When executing query:
+      """
+      MATCH (a)
+      CREATE (a)-[:KNOWS]->(b {name: missing})
+      RETURN b
+      """
+    Then a SyntaxError should be raised at compile time: UndefinedVariable
+
+  Scenario: Failing when using undefined variable in SET
+    When executing query:
+      """
+      MATCH (a)
+      SET a.name = missing
+      RETURN a
+      """
+    Then a SyntaxError should be raised at compile time: UndefinedVariable
+
+  Scenario: Failing when using undefined variable in DELETE
+    When executing query:
+      """
+      MATCH (a)
+      DELETE x
+      """
+    Then a SyntaxError should be raised at compile time: UndefinedVariable
+
+  Scenario: Failing when using already bound variable in CREATE
```

Comment:
"when using already bound variable" -> "when using a variable that is already bound"